client-misk/src/test/kotlin/app/cash/backfila/client/misk/base/UnshardedHibernateBackfillTest.kt

```diff
@@ -0,0 +1,272 @@
+package app.cash.backfila.client.misk.base
+
+import app.cash.backfila.client.misk.ClientMiskService
+import app.cash.backfila.client.misk.ClientMiskTestingModule
+import app.cash.backfila.client.misk.DbMenu
+import app.cash.backfila.client.misk.EmbeddedBackfila
+import app.cash.backfila.client.misk.EmbeddedBackfillRun
+import app.cash.backfila.client.misk.MenuQuery
+import com.google.inject.Module
+import com.google.inject.TypeLiteral
+import misk.hibernate.Id
+import misk.hibernate.Query
+import misk.hibernate.Session
+import misk.hibernate.Transacter
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import okhttp3.internal.toImmutableList
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+@MiskTest(startService = true)
+class UnshardedHibernateBackfillTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module: Module = ClientMiskTestingModule(false, listOf(TestBackfill::class))
+
+  @Inject @ClientMiskService lateinit var transacter: Transacter
+  @Inject lateinit var embeddedBackfila: EmbeddedBackfila
+
+  class TestBackfill @Inject constructor(
+    entityType: TypeLiteral<DbMenu>,
```

Comment:
is there a way to not have to provide these in every backfill?

Reply:
Yeah I'm thinking about it. Not sure if there is a better way to approach this problem. Might be one with Kotlin. I'll investigate this later though.

Reply:
Yeah the new code makes this go away!